OpenRA.Mods.Common/TraitsInterfaces.cs

```diff
@@ -126,4 +126,17 @@ public interface IPreventsAutoTarget
 	{
 		bool PreventsAutoTarget(Actor self, Actor attacker);
 	}
+
+	[Flags]
+	public enum UncloakType
+	{
+		None = 0,
+		Attack = 1,
+		Move = 2,
+		Unload = 4,
+		Infiltrate = 8,
+		Demolish = 16,
+		Damage = 32,
+		Default = Attack | Unload | Infiltrate | Demolish
```

Comment:
Nice!  We can build that into the docs generation and then remove the (IMO ugly) cases where we list the enum values in the [Desc].